pkgs/development/compilers/gcc/12/default.nix

```diff
@@ -75,10 +74,10 @@ let majorVersion = "12";
         sha256 = "sha256-JqCGJAfbOxSmkNyq49aFHteK/RFsCSLQrL9mzUCnaD0=";
       })
       ++ optional langD ../libphobos.patch
-      ++ optionals (targetPlatform.libc == "serenitylibc") (
-        (builtins.filter (lib.hasSuffix ".patch") (lib.filesystem.listFilesRecursive "${serenity.bootstrap-source}/Toolchain/Patches/gcc"))
-        ++ [ ./serenity-libtool-enable-shared-library-support.patch ]
-      )
+      ++ optionals (targetPlatform.libc == "serenitylibc") [
+        ./serenity-support.patch
+        ./serenity-libtool-enable-shared-library-support.patch
```

Comment:
cc @aakropotkin for the libtool-related patch

Reply:
I'm on mobile and can't view the whole file just the diff.

Just be sure this is in a conditional block tied to the libtool version so it doesn't need to be patched again when this is applied upstream in a new release 🤘

Reply:
> Just be sure this is in a conditional block tied to the libtool version so it doesn't need to be patched again when this is applied upstream in a new release metal

I think we don't do that in nixpkgs. Probably mainly so we don't carry around patches that aren't needed anymore.
We often apply patches unconditionally, even though they'd only be needed on a specific platform, for the same reason.

Reply:
I just read the whole thing and it looks good. There's a chance Serenity users will have issues when libtool 2.8 comes out but that's less destructive than everything being broken.

You could save add a version check to the condition for extra coverage.